src/tests/managed_array_tests.cpp

```diff
@@ -409,4 +409,40 @@ CUDA_TEST(ManagedArray, Move)
 
   array.free();
 }
+
+CUDA_TEST(ManagedArray, MigrateInnerData)
+{
+  chai::ManagedArray<chai::ManagedArray<int>> gpuArray(3, chai::GPU);
+
+  for (int i = 0; i < 3; ++i)
+  {
+     gpuArray[i] = chai::ManagedArray<int>(3, chai::GPU);
+  }
+
+  forall(cuda(), 0, 3, [=] __device__ (int i) {
+      gpuArray[0][i] = i;
+      gpuArray[1][i] = i;
+      gpuArray[2][i] = i;
+  });
+
+  auto hostArray = chai::ManagedArray<chai::ManagedArray<int>>(gpuArray);
+
+  for (int i = 0; i < 3; ++i)
```

Comment:
Make this loop look like the one above, but use sequential instead of cuda. You can have multiple ASSERT statements inside the loop.

Reply:
OK, I think I have a good test case now. I did notice that only shallow copies are being done currently. Is that correct?